src/Domain/CoordinateMaps/TimeDependent/ShapeTransitionFunctions.hpp

```diff
@@ -0,0 +1,47 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/Protocols.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#pragma once
+
+namespace PUP {
+class er;
+}  // namespace PUP
+
+namespace domain::CoordinateMaps {
+/*!
+ * \ingroup CoordMapsTimeDependentGroup
+ * \brief A transition function that falls off as \f$f(r) = 1 / r\f$.
+ */
+class SphereTransition : public tt::ConformsTo<protocols::TransitionFunc> {
+ public:
+  template <typename T>
+  T operator()(std::array<T, 3> coords) const {
+    return 1. / magnitude(coords);
+  }
+
+  template <typename T>
+  T map_over_radius(std::array<T, 3> coords) const {
+    const auto one_over_magnitude = 1. / magnitude(coords);
+    return one_over_magnitude * one_over_magnitude;
+  }
+
+  template <typename T>
+  std::array<T, 3> gradient(std::array<T, 3> coords) const {
+    const auto one_over_magnitude = 1. / magnitude(coords);
+    return -coords * one_over_magnitude * one_over_magnitude *
+           one_over_magnitude;
```

Comment:
Maybe use `cube` in `Utilities/ConstantExpressions.hpp` ?  And `square` for `map_over_radius`.